<launch>

    <!-- params required by navigation.py and auto.py -->
    <rosparam param="/core_rover/Mode">"Auto"</rosparam>
    <rosparam param="rpm_limit">1000</rosparam>
    <rosparam param="steer_limit">1000</rosparam>

    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization_node">

        <!-- node configuration -->
        <param name="frequency" value="10"/>
        <param name="sensor_timeout" value="0.2"/>

        <param name="map_frame" value="map"/>
        <param name="odom_frame" value="odom"/>
        <param name="base_link_frame" value="base_link"/>
        <param name="world_frame" value="odom"/>

        <!-- subscription to IMU topic -->
        <param name="imu0" value="/imu_data"/>
        <rosparam param="imu0_config">[false,false,false,
                                       false,true,false,
                                       false,false,false,
                                       false,false,false,
                                       false,false,false]</rosparam>

        <!-- publish to filtered odometry -->
        <remap from="odometry/filtered" to="/odom_filtered"/>

    </node>

    <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node">
       
        <!-- read raw imu, raw gps, and filtered odometry -->
        <remap from="imu/data" to="/imu_data"/>
        <remap from="gps/fix" to="/gps_data"/>
        <remap from="odometry/filtered" to="/odom_filtered"/>

        <!-- publish to filtered gps -->
        <param name="publish_filtered_gps" value="true"/>
        <remap from="gps/filtered" to="/gps_filtered"/>

    </node>

</launch>
